### PR TITLE
Check for SEACAS support.

### PR DESCRIPTION
**Pull request by drwells**
We need it for ExodusII.

@tamiko approved the CMake changes in #10899 so we might as well merge it in.

### CODE REVIEW

File: cmake/configure/configure_2_trilinos.cmake [hunk 1]
```diff
@@ -281,6 +281,43 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
...
+      CHECK_CXX_SOURCE_COMPILES(
+        "
+        #include <exodusII.h>
+        int
+        main()
+        {
+          int component_word_size = sizeof(double);
+          int floating_point_word_size = 0;
+          float ex_version = 0;
+          const int ex_id = ex_open(\"test.ex\",
```
Comment (bangerth) [inline cmake/configure/configure_2_trilinos.cmake:298]: I wished Trilinos would pay more attention to namespacing :-(

File: cmake/configure/configure_2_trilinos.cmake [hunk 2]
```diff
@@ -171,7 +171,7 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
       #
       # Check for modules.
       #
-      FOREACH(_optional_module EpetraExt ROL Sacado Tpetra MueLu Zoltan)
+      FOREACH(_optional_module EpetraExt MueLu ROL Sacado SEACAS Tpetra Zoltan)
```
Comment (bangerth) [inline cmake/configure/configure_2_trilinos.cmake:174]: Does this mean that SEACAS is not a *required* component of Trilinos? Do we need to adjust things such as candi and our installation documentation, or is SEACAS a default component of Trilinos?

File: cmake/configure/configure_2_trilinos.cmake (hunk 1 above)
Comment (drwells) [reply cmake/configure/configure_2_trilinos.cmake:298]: To be fair Exodus, AFAICT, very much predates Trilinos (some of the technical documents are from the early 1990s).

**Comment by drwells** [comment]
Let's not split this off from #10899 now that that PR is feature-complete.

**Pull request closed without merging**